src/lib/optimizer/strategy/dependent_group_by_reduction_rule.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include "abstract_rule.hpp"
+#include "expression/abstract_expression.hpp"
+#include "expression/lqp_column_expression.hpp"
+
+namespace opossum {
+
+class AbstractLQPNode;
+
+// Removes columns from list of group by columns when they are functionally dependent from column(s) of the grouping.
+// This is currently only the case, when both the primary key and other columns of the same table are grouped.
+// This is regularly the case, when the columns are supposed to be later accessed (e.g., see usage of "c_acctbal" in
+// TPC-H query 10). As "c_acctbal" does not need to be in the grouping (since the primary key c_custkey is present),
```

Comment:
The "this is regularly..." sentence is a bit confusing. Rather give an example and describe it?

Reply:
Added an example.